### PR TITLE
Fix Color widget delete exception

### PR DESCRIPTION
**Pull request by iamved**

### CODE REVIEW

File: Orange/widgets/data/owcolor.py [hunk 1]
```diff
@@ -298,12 +298,16 @@ def create_part(variables):
 
     def storeSpecificSettings(self):
         # Store the colors that were changed -- but not others
-        self.current_context.disc_data = \
-            [(var.name, var.values, "colors" in var.attributes and var.colors)
-             for var in self.disc_colors]
-        self.current_context.cont_data = \
-            [(var.name, "colors" in var.attributes and var.colors)
-             for var in self.cont_colors]
+        self.closeContext()
```
Comment (ales-erjavec) [inline Orange/widgets/data/owcolor.py]: `storeSpecificSettings()` is called from from within `closeContext()` so this causes an infinite recursion.

File: Orange/widgets/data/owcolor.py [hunk 2]
```diff
@@ -298,12 +298,16 @@ def create_part(variables):
...
+        if self.current_context is not None:    
+            self.openContext(disc_data)
```
Comment (ales-erjavec) [inline Orange/widgets/data/owcolor.py]: `disc_data` is not defined in this scope.

File: Orange/widgets/data/owcolor.py [hunk 3]
```diff
@@ -298,12 +298,16 @@ def create_part(variables):
...
+            self.current_context.disc_data = \
+                [(var.name, var.values, "colors" in var.attributes and var.colors)
+                 for var in self.disc_colors]
+            self.openContext(cont_data)
```
Comment (ales-erjavec) [inline Orange/widgets/data/owcolor.py]: `cont_data` is not defined in this scope

**Comment by ales-erjavec** [comment]
`close/openContext` should not be called when storing/loading settings. They are only called when the widget receives new inputs. They store/restore the settings based on previously seen domains.

**Comment by iamved** [comment]
Thanks for letting me know. Making required changes.

File: Orange/widgets/data/owcolor.py (hunk 1 above)
Comment (iamved) [inline Orange/widgets/data/owcolor.py]: Please explain me what kind of infinite recursion it will cause?

**Comment by iamved** [comment]
` def storeSpecificSettings(self):
        # Store the colors that were changed -- but not others
        self.disc_data =[]
        self.cont_data=[]
        if self.current_context is not None:    
            self.openContext(disc_data)
            self.current_context.disc_data = \
                [(var.name, var.values, "colors" in var.attributes and var.colors)
                 for var in self.disc_colors]
            self.openContext(cont_data)
            self.current_context.cont_data = \
                [(var.name, "colors" in var.attributes and var.colors)
                 for var in self.cont_colors]`

File: Orange/widgets/data/owcolor.py (hunk 1 above)
Comment (ales-erjavec) [inline Orange/widgets/data/owcolor.py]:
```
--------------------------------------------------------------------------------
RecursionError                                Traceback (most recent call last):
  File "/Users/aleserjavec/workspace/orange3/Orange/canvas/scheme/widgetsscheme.py", line 403, in eventFilter
    widget.saveSettings()
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/widget.py", line 468, in saveSettings
    self.settingsHandler.update_defaults(self)
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/settings.py", line 447, in update_defaults
    self.settings_from_widget(widget)
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/settings.py", line 554, in settings_from_widget
    widget.storeSpecificSettings()
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/data/owcolor.py", line 301, in storeSpecificSettings
    self.closeContext()
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/widget.py", line 459, in closeContext
    self.settingsHandler.close_context(self)
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/settings.py", line 543, in close_context
    self.settings_from_widget(widget)
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/settings.py", line 554, in settings_from_widget
    widget.storeSpecificSettings()
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/data/owcolor.py", line 301, in storeSpecificSettings
....
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/data/owcolor.py", line 301, in storeSpecificSettings
    self.closeContext()
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/widget.py", line 459, in closeContext
    self.settingsHandler.close_context(self)
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/settings.py", line 543, in close_context
    self.settings_from_widget(widget)
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/settings.py", line 554, in settings_from_widget
    widget.storeSpecificSettings()
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/data/owcolor.py", line 301, in storeSpecificSettings
    self.closeContext()
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/widget.py", line 459, in closeContext
    self.settingsHandler.close_context(self)
  File "/Users/aleserjavec/workspace/orange3/Orange/widgets/settings.py", line 543, in close_context
    self.settings_from_widget(widget)
RecursionError: maximum recursion depth exceeded
```

**Comment by iamved** [comment]
I hope deleting `closecontext()` and adding defining `disc_data`, `cont_data` in scope of `storeSpecificSetting()` will make it error free. If this is okay then I will make a new PR for the issue.

**Comment by ales-erjavec** [comment]
Please do not create new PR. Commit/amend your changes on this branch and (force) push to your github clone (to the same branch).
The pull request will be updated.

**Comment by iamved** [comment]
Please check the updates that I have been made.

**Comment by astaric** [comment]
Thank you for the pull request. In the end, we have decided for a more general fix for the issue (see  #1104) so I am closing this pull request.

**Pull request closed without merging**